agent/taskresource/envFiles/envfile.go

```diff
@@ -0,0 +1,198 @@
+package envFiles
+
+import (
+	"sync"
+	"time"
+
+	"path/filepath"
+	"strings"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	"github.com/aws/amazon-ecs-agent/agent/s3"
+	"github.com/aws/amazon-ecs-agent/agent/s3/factory"
+	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
+	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+)
+
+const (
+	envFileDirPath    = "envfiles"
+	envTempFilePrefix = "tmp_env"
+
+	s3DownloadTimeout = 30 * time.Second
+)
+
+// represents envfile as a task resource
```

Comment:
comment should begin with the name https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences. i.e. `// EnvironmentFileResource ..`

Reply:
done